### PR TITLE
Adjust inmem to example schema

### PR DESCRIPTION
**Pull request by wtrocki**
New updates for the schema that involve relationships.
Supports both inmemory and posgress updates.

## Notes

Now it is not possible to create meme without Profile - user needs to be created first.

### Memeolist - In memory 

To start the application with MemeoList schema and queries with an in-memory data source, run these commands:
```
docker-compose -p aerogeardatasyncserver up
npm run db:init:memeo:inmem
npm run dev:memeo
```

### Memeolist - Postgres 

To start the application with MemeoList schema and queries with an Postgres source, run these commands:
```
docker-compose -p aerogeardatasyncserver up
npm run db:init:memeo:postgres
npm run dev:memeo
``` 

Make sure that you remove previous schema by dropping containers with postgres sql.

### CODE REVIEW

File: examples/memeolist.query.graphql [hunk 1]
```diff
@@ -2,12 +2,52 @@ query allMemes{
...
+    owner
+    likes
+    comments {
+      id
+    }
+  }
+}
+
+query profile {
+  profile(email:"wtr@redhit.com"){
```
Comment (aliok) [inline examples/memeolist.query.graphql]: redhit :D

**Comment by wtrocki** [comment]
Ready for review!

File: server/lib/resolvers/builders/nedb.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ function buildNeDBResolver (dataSource, compiledRequestMapping, compiledResponse
       const parsedQuery = JSONParse(queryString)
 
       if (parsedQuery.error) {
+        console.error('Error when parsing query', queryString, parsedQuery.error)
```
Comment (aliok) [inline server/lib/resolvers/builders/nedb.js]: Please use "log" like in other modules. There's actually a linter rule that prevents using console

**Comment by aliok** [comment]
Some problems:
* `likeMeme` returns true even when the meme doesn't exist: Both inmem and Postgres
* Same thing above, but for `postComment` and only for inmem

@wtrocki  if you have the energy to fix them it would be nice.

File: server/lib/resolvers/builders/nedb.js (hunk 2 above)
Comment (aliok) [reply server/lib/resolvers/builders/nedb.js]: I wonder why that linter didn't complain...

File: examples/memeolist.tables.sql [hunk 3]
```diff
@@ -1,6 +1,21 @@
-DROP TABLE IF EXISTS "Meme";
+CREATE TABLE Profile (
```
Comment (darahayes) [inline examples/memeolist.tables.sql]: Can we add back in the following here:

```
DROP TABLE IF EXISTS "Meme";
DROP TABLE IF EXISTS "Profile";
DROP TABLE IF EXISTS "Comment";
```

It just makes local dev a lot simpler

File: server/lib/resolvers/builders/nedb.js [hunk 4]
```diff
@@ -48,6 +49,7 @@ function buildNeDBResolver (dataSource, compiledRequestMapping, compiledResponse
 
       function mapResponse (err, res) {
         if (err) return reject(err)
+        if (!res) return resolve()
```
Comment (camilamacedo86) [inline server/lib/resolvers/builders/nedb.js]: @aliok should not the linter complain about this point either?

Comment (aliok) [reply server/lib/resolvers/builders/nedb.js]: @camilamacedo86 I rebased this branch on top of master (has now linter config fixed)  *locally* and it didn't complain about that line. What's the problem there? BTW, we're using this format: https://standardjs.com

**Comment by wtrocki** [comment]
> likeMeme returns true even when the meme doesn't exist: Both inmem and Postgres

I really struggled to get this implemented in the current resolver format. 
Normally I would fetch the Meme, increment number of the likes and save it. Return true or false.
In the template I cannot execute two queries at the same time or have logic like:

```
if(numerOfRowsAffected ===0) return false;
```

> Same thing above, but for postComment and only for inmem

That should be easy fix.

**Comment by aliok** [comment]
ok, good to merge then.

**Pull request merged**